### DISCUSSION TITLE
Cannot build sub module separately with Tycho 2.7.3

### DISCUSSION THREAD

**Discussion by selundqma** [eclipse-tycho/tycho#1055] [Q&A] [answered]
Hi!

I have a multi-module Maven project spread over 8 repos/gits to build my feature-based RCP application. Everything can be built fine with 2.6.0, but with 2.7.3 I can only build the complete project from the parent POM. I can no longer build a single module/git without getting an error like this:

[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:2.7.3:validate-classpath (default-validate-classpath) on project com.xxxxxxxxxxx.testclient: Execution default-validate-classpath of goal org.eclipse.tycho:tycho-compiler-plugin:2.7.3:validate-classpath failed: Exception parsing OSGi MANIFEST C:\m2\repository\com\xxxxxxxxxxx\yyyyyyyyyyyy\client\client-common\com.xxxxxxxxx.client.common.feature\22.7.1-SNAPSHOT\com.xxxxxxxxxxx.yyyyyyyyyyyy.client.common.feature-22.7.1-SNAPSHOT.jar!/META-INF/MANIFEST.MF: MANIFEST header 'Bundle-Version' not found -> [Help 1]

`<pomDependencies>consider</pomDependencies>` is used in the parent POM.

So.. why does tycho suddenly try to read "Bundle-Version" from the MANIFEST.MF file of an Eclipse feature JAR??? There is no such information in that Tycho auto-generated file. I wouldn´t even know how to get that information into that feature JAR.

These rows are present in the POM file in the sub module I am trying to build separately.

        <dependency>
            <groupId>com.xxxxxxxxxxx.yyyyyyyyyyyy.client.client-common</groupId>
            <artifactId>com.xxxxxxxxxxx.client.common.feature</artifactId>
            <version>22.7.1-SNAPSHOT</version>
            <type>eclipse-feature</type>
        </dependency>
        <dependency>
            <groupId>com.xxxxxxxxxxx.yyyyyyyyyyyy.client.client-common</groupId>
            <artifactId>com.xxxxxxxxxxx.client.common.feature</artifactId>
            <version>22.7.1-SNAPSHOT</version>
            <classifier>p2metadata</classifier>
            <type>xml</type>
        </dependency>
        <dependency>
            <groupId>com.xxxxxxxxxxx.yyyyyyyyyyyy.client.client-common</groupId>
            <artifactId>com.xxxxxxxxxxx.client.common.feature</artifactId>
            <version>22.7.1-SNAPSHOT</version>
            <classifier>p2artifacts</classifier>
            <type>xml</type>
        </dependency>

As I said, this worked fine until moving on to version 2.7.3. Any ideas what is wrong?

**Comment by laeubi** [comment]
> As I said, this worked fine until moving on to version 2.7.3. Any ideas what is wrong?

Most probably there is no test covering this use-case you you like to provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue?

**Reply by selundqma** [reply]
Hi laeubi! I have created a pull request including an example project to demonstrate the issue.

https://github.com/eclipse/tycho/pull/1057

BR, Mats

**Comment by selundqma** [comment, marked as answer]
Hi @laeubi !!

Thank you very much for all help and the fix! I have tested a local build now and IT WORKS with 2.7.4-SNAPSHOT!! Sending you a virtual flower :o).

BR, Mats (selundqma)
